sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala

```diff
@@ -151,7 +151,7 @@ abstract class NumericType extends AtomicType {
 }
 
 
-private[sql] object NumericType extends AbstractDataType {
+private[spark] object NumericType extends AbstractDataType {
```

Comment:
I don't think it's required - see https://github.com/apache/spark/pull/9777/files#diff-a364fbaaec2088b49d5d1ceaea4a1c41R73 where OneHotEncoder uses `isInstanceOf[NumericType]` without making this object `private[spark]`. Here `NumericType` refers to the abstract class [here](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala#L144).